scipy/stats/morestats.py

```diff
@@ -213,6 +215,15 @@ def kstat(data, n=2):
     S = zeros(n + 1, 'd')
     data = ravel(data)
     N = len(data)
+
+    # raise ValueError on empty input
+    if N == 0:
+        raise ValueError("Data input must not be empty")
+
+    # on nan input, return nan without warning
+    if np.isnan(np.sum(data)):
+        return np.nan
+
     for k in range(1, n + 1):
         S[k] = sum(data**k, axis=0)
```

Comment:
This ``sum`` is ugly, is actually ``np.sum`` which gets imported as ``from numpy import sum`` that shadows the builtin ``sum``. Would be good to fix that in the whole file.